AdminServer/appscale/admin/__init__.py

```diff
@@ -952,6 +976,109 @@ def get(self, project_id, operation_id):
     self.write(json_encode(operation.rest_repr()))
 
 
+class OAuthHandler(BaseHandler):
+  """ Authorize users and give them Access Tokens."""
+  def initialize(self, ua_client):
+    """ Defines required resources to handle requests.
+
+    Args:
+      ua_client: A UAClient.
+    """
+    self.ua_client = ua_client
+
+  def post(self):
+    """Grants users Access Tokens."""
+
+    # Format for error message.
+    error_msg = {
+      'error': '',
+      'error_description': ''
+    }
+
+    missing_arguments = []
+
+    grant_type = self.get_argument('grant_type', default=None, strip=True)
+    if not grant_type:
+      missing_arguments.append('grant_type')
+
+    username = self.get_argument('username', default=None, strip=True)
+    if not username:
+      missing_arguments.append('username')
+
+    password = self.get_argument('password', default=None, strip=True)
+    if not password:
+      missing_arguments.append('password')
+
+    if missing_arguments:
```

Comment:
The client should specify the scope when requesting the token. That way, users with different privileges can request different scopes.

Reply:
scope is technically an optional argument though isn't it? and do we want to accept whichever scope they specify?

Reply:
We should only allow the single scope for now. It would be optional if we didn't anticipate requiring different scopes for different actions in the future. Without requiring a scope, we won't know what permissions the client is asking for.

Clients that depend on this functionality would need to change their behavior if we wait until later to make that a required field.